<a href="https://colab.research.google.com/github/najla-alhusaini/it326-telco-churn/blob/main/Phase1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

url = 'https://raw.githubusercontent.com/najla-alhusaini/it326-telco-churn/main/Dataset/Raw_dataset.csv'
df = pd.read_csv(url)

print("Shape (rows, cols):", df.shape)
print("\nData types:\n", df.dtypes)

TARGET_COLUMN = 'Churn'
if TARGET_COLUMN in df.columns:
    print("\nClass counts:")
    print(df[TARGET_COLUMN].value_counts())
else:
    print(f"Target column '{TARGET_COLUMN}' not found!")

print("\nSample of dataset:")
display(df.head())

Shape (rows, cols): (7043, 21)

Data types:
 customerID           object
gender               object
SeniorCitizen         int64
Partner              object
Dependents           object
tenure                int64
PhoneService         object
MultipleLines        object
InternetService      object
OnlineSecurity       object
OnlineBackup         object
DeviceProtection     object
TechSupport          object
StreamingTV          object
StreamingMovies      object
Contract             object
PaperlessBilling     object
PaymentMethod        object
MonthlyCharges      float64
TotalCharges         object
Churn                object
dtype: object

Class counts:
Churn
No     5174
Yes    1869
Name: count, dtype: int64

Sample of dataset:


customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

# Phase 1 - Dataset Overview

## Project Goal
The goal of this project is to **classify telecom customers** based on whether they will churn (leave the company) or not.  
- **Classification Task:** Predict the target attribute `Churn` (Yes / No) using customer demographic, service, and billing information.  
- **Clustering Task:** Group customers into segments based on their attributes to discover patterns in customer behavior and identify groups with higher churn risk.
- These tasks help telecom companies identify at-risk customers and design targeted retention strategies for different customer segments
## Dataset Source
- **Source:** Kaggle – [Telco Customer Churn Dataset](https://www.kaggle.com/datasets/blastchar/telco-customer-churn)  
- **File:** `Dataset/Raw_dataset.csv`

## Dataset Description
- The dataset contains 21 attributes with mixed data types: categorical (e.g., gender, Contract, PaymentMethod), numerical (SeniorCitizen, tenure, MonthlyCharges), and one object-type column (TotalCharges) that requires conversion to numeric due to non-standard entries.

- It includes 7,043 records (instances) representing customer information.

- The class attribute is Churn, which indicates whether a customer left the company. The distribution is:

	•	No (Non-churn): 5,174 instances
	•	Yes (Churn): 1,869 instances


